datasetinsights/visualization/object_detection.py

```diff
@@ -8,6 +9,140 @@
 from .plots import histogram_plot, rotation_plot
 
 
+class ScaleFactor:
+    """Generate scale factor distribution.
+
+    Scale Factor describes the size of the rendered object in a capture.
+    Higher the scale factor, higher would be the visible pixels.
+
+    Atrributes:
+        captures(sim.Captures): a collection of capture records.
+
+    """
+
+    def __init__(self, data_root):
+        """
+
+        Args:
+            data_root(str): path to the dataset.
+
+        """
+        capture = sim.Captures(data_root)
+        self.captures = capture.filter(
+            def_id=constants.BOUNDING_BOX_2D_DEFINITION_ID
+        )
+
+    @staticmethod
+    def generate_scale_data(captures):
+        """ Method to extract scale parameter from sensor data.
+
+        Args:
+            captures(sim.Captures): a collection of capture records.
+
+        Returns:
+            list: extracted 'scale' parameter from the sensor data.
```

Comment:
Fix docstring here. The return value is a pandas dataframe here?

Reply:
Fixed.